Server/Web/WebServer.cs

```diff
@@ -65,6 +65,18 @@ public static void StartWebServer()
                         Server.Use(new ExceptionHandler());
                         Server.Use(new CompressionHandler(DeflateCompressor.Default, GZipCompressor.Default));
                         Server.Use(new HttpRouter().With(string.Empty, new RestHandler<ServerInformation>(new ServerInformationRestController(), JsonResponseProvider.Default)));
+
+                        // If the Prometheus endpoint is enabled, build and add the HttpRouter to handle it to the server.
+                        if (MetricsSettings.SettingsStore.Enabled) {
+                            LunaLog.Info("Enabling Prometheus endpoint");
+                            Server.Use(
+                                new HttpRouter().With(
+                                    MetricsSettings.SettingsStore.Endpoint,
+                                    new MetricsHandler()
+                                )
+                            );
+                        }
```

Comment:
Instead of creating a separate `HttpRouter` for this path, I think it would be a bit cleaner to add the metrics sub-handler to the existing one (which does require a bit of restructuring due to the if-clause), something like this?

(`HttpRouter.With()` does modify itself, so we are fine ignoring the returned object).

```suggestion
                        var httpRouter = new HttpRouter();
                        Server.Use(httpRouter);
                        httpRouter.With(string.Empty, new RestHandler<ServerInformation>(new ServerInformationRestController(), JsonResponseProvider.Default));

                        // If the Prometheus endpoint is enabled, add it to the HttpRouter.
                        if (MetricsSettings.SettingsStore.Enabled) {
                            LunaLog.Info("Enabling Prometheus endpoint");
                            httpRouter.With(
                                MetricsSettings.SettingsStore.Endpoint,
                                new MetricsHandler()
                            );
                        }
```

Reply:
Yeah, it would be cleaner if we could avoid the creation of another HttpRouter object.